homeassistant/components/twitch/sensor.py

```diff
@@ -176,23 +137,20 @@ def update(self):
                 raise Exception("Unknown error response on check_user_subscription")
 
             follows = self._client.get_users_follows(
-                from_id=user["id"], to_id=self._channel["id"]
+                from_id=user, to_id=self.unique_id
             )["data"]
-            if len(follows) > 0:
-                self._follow = {
-                    ATTR_FOLLOW: True,
-                    ATTR_FOLLOW_SINCE: follows[0]["followed_at"],
-                }
-            else:
-                self._follow = {ATTR_FOLLOW: False}
+            self._attr_extra_state_attributes[ATTR_FOLLOW] = len(follows) > 0
+            if len(follows):
+                self._attr_extra_state_attributes[ATTR_FOLLOW_SINCE] = follows[0][
+                    "followed_at"
+                ]
 
-        streams = self._client.get_streams(user_id=[self._channel["id"]])["data"]
-        if len(streams) > 0:
+        if streams := self._client.get_streams(user_id=[self.unique_id])["data"]:
             stream = streams[0]
-            self._game = stream["game_name"]
-            self._title = stream["title"]
-            self._preview = stream["thumbnail_url"]
-            self._state = STATE_STREAMING
+            self._attr_native_value = STATE_STREAMING
+            self._attr_extra_state_attributes[ATTR_GAME] = stream["game_name"]
+            self._attr_extra_state_attributes[ATTR_TITLE] = stream["title"]
```

Comment:
It seems like these are not reset in the else construct

Reply:
(when not streaming it cannot provide a game + title?)

Reply:
Check L115, we are cleaning out any old values there. And yes, no game or title when not streaming.

Reply:
That removes the attributes completely, which is not correct.
The attributes should stick around (but have a `None` value).

For context see: <https://github.com/home-assistant/architecture/discussions/680>

Reply:
Thanks. I was not aware of that decision.